File: Objective-C/UIKit/UIButton/UIButton+Title.m
//
//  UIButton+Title.m
//  UIButtonTest
//
//  Created by AngieMIta on 2017/8/23.
//  Copyright © 2017年 crazyit.org. All rights reserved.
//

#import "UIButton+Title.h"

@implementation UIButton (Title)


-(NSString *)titleForNormal{
    return [self titleForState:UIControlStateNormal];
}

-(NSString *)titleForHighlighted{
    return [self titleForState:UIControlStateHighlighted];
}

-(NSString *)titleForDisabled{
    return [self titleForState:UIControlStateDisabled];
}

-(NSString *)titleForSelected{
    return [self titleForState:UIControlStateSelected];
}


- (void)setTitleForNormal:(nullable NSString *)title{
    [self setTitle:title forState:UIControlStateNormal];
}

- (void)setTitleForHighlighted:(nullable NSString *)title{
    [self setTitle:title forState:UIControlStateHighlighted];
}

- (void)setTitleForDisabled:(nullable NSString *)title{
    [self setTitle:title forState:UIControlStateDisabled];
}

- (void)setTitleForSelected:(nullable NSString *)title{
    [self setTitle:title forState:UIControlStateSelected];
}

@end

File: Swift/UIKit/UIButton+Title.swift
//
//  UIButton+title.swift
//  vshiMessge
//
//  Created by 巴糖 on 2017/7/28.
//  Copyright © 2017年 巴糖. All rights reserved.
//

import UIKit

// MARK: - Properties
public extension UIButton {
    
    @IBInspectable
    /// Title of normal state for button; also inspectable from Storyboard.
    public var titleForNormal: String? {
        get {
            return title(for: .normal)
        }
        set {
            setTitle(newValue, for: .normal)
        }
    }
    
    @IBInspectable
    /// Title of highlighted state for button; also inspectable from Storyboard.
    public var titleForHighlighted: String? {
        get {
            return title(for: .highlighted)
        }
        set {
            setTitle(newValue, for: .highlighted)
        }
    }
    
    @IBInspectable
    /// Title of selected state for button; also inspectable from Storyboard.
    public var titleForSelected: String? {
        get {
            return title(for: .selected)
        }
        set {
            setTitle(newValue, for: .selected)
        }
    }
    
    @IBInspectable
    /// Title of disabled state for button; also inspectable from Storyboard.
    public var titleForDisabled: String? {
        get {
            return title(for: .disabled)
        }
        set {
            setTitle(newValue, for: .disabled)
        }
    }
}
